.github/workflows/ci.yml

```diff
@@ -0,0 +1,84 @@
+name: CI
```

Comment:
Let's keep the workflow in the CICD.yml file

Reply:
Hm, my plan was to later add a separate `deploy.yml` workflow with the compile steps for release binaries. Wanted to add them as separate workflow to not have one huge setting but rather smaller purpose built files.

Would you like to have everything in one file instead?

Reply:
Yes for now I'd prefer a single file. We can split it later if we find it is becoming too large and hard to read.